qcodes/dataset/database_copy_paste.py

```diff
@@ -0,0 +1,397 @@
+        cursor.execute(layout_insert, values)
+        source_layout_ids.append(row['layout_id'])
+        target_layout_ids.append(cursor.lastrowid)
+
+    # for the dependencies, we need a map from source layout_id to
+    # target layout_id
+    layout_id_map = dict(zip(source_layout_ids, target_layout_ids))
+
+    placeholders = sql_placeholder_string(len(source_layout_ids))
+
+    deps_query = f"""
+                 SELECT dependent, independent, axis_num
+                 FROM dependencies
+                 WHERE dependent IN {placeholders}
+                 OR independent IN {placeholders}
+                 """
+
+    cursor = source_conn.cursor()
+    cursor.execute(deps_query, tuple(source_layout_ids*2))
+    rows = cursor.fetchall()
+
+    deps_insert = """
+                  INSERT INTO dependencies
+                  (dependent, independent, axis_num)
+                  VALUES (?,?,?)
+                  """
+    cursor = target_conn.cursor()
+
+    for row in rows:
+        values = (layout_id_map[row['dependent']],
+                  layout_id_map[row['independent']],
+                  row['axis_num'])
+        cursor.execute(deps_insert, values)
+
+
+def _copy_results_table(source_conn: SomeConnection,
+                        target_conn: SomeConnection,
+                        source_run_id: int,
+                        target_run_id: int,
+                        column_names_and_types: str) -> None:
+    """
+    Copy the contents of the results table. Creates a new results_table with
+    a name appropriate for the target DB and updates the rows of that table
+    """
+    table_name_query = """
+                       SELECT result_table_name, name
+                       FROM runs
+                       WHERE run_id = ?
+                       """
+    cursor = source_conn.cursor()
+    cursor.execute(table_name_query, (source_run_id,))
+    row = cursor.fetchall()[0]
+    table_name = row['result_table_name']
+    run_name = row['name']
+
+    format_string_query = """
+                          SELECT format_string, run_counter, experiments.exp_id
+                          FROM experiments
+                          INNER JOIN runs
+                          ON runs.exp_id = experiments.exp_id
+                          WHERE runs.run_id = ?
+                          """
+    cursor = target_conn.cursor()
+    cursor.execute(format_string_query, (target_run_id,))
+    row = cursor.fetchall()[0]
+    format_string = row['format_string']
+    run_counter = row['run_counter']
+    exp_id = int(row['exp_id'])
+
+    get_data_query = f"""
+                     SELECT *
+                     FROM "{table_name}"
+                     """
+    cursor = source_conn.cursor()
+    cursor.execute(get_data_query)
+    data_rows = cursor.fetchall()
+    data_columns = data_rows[0].keys()
+    data_columns.remove('id')
+
+    target_table_name = format_table_name(format_string,
+                                          run_name,
```

Comment:
But... that's the same name?

Reply:
well, it should be :)

yeah, you're right, it's not a bug (thankfully). if the whole ensemble of functions works as expected, this won't be a deal. I just have a feeling that if smth does not work as expected, then it might be difficult to debug/read because this is an implication of the code of this function, and it feels to me a bit sneaky.

the best solution would be to pass the proper results table name as an argument so that this function only does data copy-pasting.

i leave it up to you.